### DISCUSSION TITLE
How to quit application?

### DISCUSSION THREAD

**Discussion by tonal** [fastapi/fastapi#3467] [Questions] [answered]
How to realize route "shupdown"?
```python
app = FastAPI()

@api.post("/shupdown")
async def kill():
    task = BackgroundTask(app.shupdown) # <???
    return {"success": True}
```

**Comment by Mause** [comment, marked as answer]
It depends how you're running your app - if you're running it inside gunicorn, you probably want [signal.pidfd_send_signal](https://docs.python.org/3/library/signal.html#signal.pidfd_send_signal) and either `TERM` or `QUIT` (see https://docs.gunicorn.org/en/stable/signals.html)
